src/main/java/smokefree/projection/InitiativeProjection.java

```diff
@@ -115,11 +154,82 @@ void on(CheckListUpdateEvent evt) {
     }
 
     public Playground playground(String id, String userId) {
-        return playgrounds.containsKey(id) ? playgrounds.get(id).getPlaygroundForUser(userId) : null;
+        Playground playground = playgrounds.containsKey(id) ? playgrounds.get(id).getPlaygroundForUser(userId) : null;
+        return playground;
     }
 
     public Progress progress() {
         return progress;
     }
 
+    /**
+     * It finds the playgrounds list which are nearby to the given GeoLocation.
+     * So it will useful to display in front-end, asking user for confirmation
+     * @param newPlaygroundLocation
+     * @param distance
+     * @return
+     */
+    public List<Playground> checkForNearByPlaygrounds(GeoLocation newPlaygroundLocation, long distance) {
+        GeodeticCalculator geoCalc = new GeodeticCalculator();
+        Ellipsoid reference = Ellipsoid.WGS84;
+        GlobalPosition newPlaygroundLocationPosition = new GlobalPosition(newPlaygroundLocation.getLat(), newPlaygroundLocation.getLng(), 0.0); // Point A
+        return playgrounds.entrySet().stream()
+                .filter( playgroundEntry -> {
+                    Playground playground = playgroundEntry.getValue();
+                    GlobalPosition playgroundPosition = new GlobalPosition(playground.getLat() , playground.getLng(), 0.0);
+                    double playgroundsDistance = geoCalc.calculateGeodeticCurve(reference, playgroundPosition, newPlaygroundLocationPosition).getEllipsoidalDistance();
+                    return  playgroundsDistance < distance ? true:false;
+                }).map(playgroundEntry -> playgroundEntry.getValue())
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * It cross-check whether System has already holding the maximum allowed playgrounds
```

Comment:
Good that you started this documentation. This is a bit clumsy English though...